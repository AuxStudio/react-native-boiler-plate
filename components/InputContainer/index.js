import React from 'react';
import PropTypes from 'prop-types';
import { View, ViewPropTypes } from 'react-native';
import { KeyboardAwareScrollView } from 'react-native-keyboard-aware-scroll-view';

import styles from './styles';

const propTypes = {
  wrapperStyle: ViewPropTypes.style,
  containerStyle: ViewPropTypes.style,
  children: PropTypes.node,
};

const defaultProps = {};

const InputContainer = ({ wrapperStyle, containerStyle, children }) => {
  return (
    <View style={[styles.container, wrapperStyle]}>
      <KeyboardAwareScrollView
        keyboardShouldPersistTaps="handled"
        contentContainerStyle={containerStyle}
      >
        {children}
      </KeyboardAwareScrollView>
    </View>
  );
};

InputContainer.propTypes = propTypes;
InputContainer.defaultProps = defaultProps;

export default InputContainer;
