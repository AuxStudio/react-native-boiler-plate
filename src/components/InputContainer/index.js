import React from 'react';
import PropTypes from 'prop-types';
import { View, ViewPropTypes } from 'react-native';
import { KeyboardAwareScrollView } from 'react-native-keyboard-aware-scroll-view';

import styles from './styles';

const propTypes = {
  containerStyle: ViewPropTypes.style,
  contentContainerStyle: ViewPropTypes.style,
  children: PropTypes.node,
};

const defaultProps = {};

const InputContainer = ({ containerStyle, contentContainerStyle, children }) => {
  return (
    <View style={[styles.container, containerStyle]}>
      <KeyboardAwareScrollView
        keyboardShouldPersistTaps="handled"
        contentContainerStyle={contentContainerStyle}
      >
        {children}
      </KeyboardAwareScrollView>
    </View>
  );
};

InputContainer.propTypes = propTypes;
InputContainer.defaultProps = defaultProps;

export default InputContainer;
