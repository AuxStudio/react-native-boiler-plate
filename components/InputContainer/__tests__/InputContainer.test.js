import React from 'react';
import renderer from 'react-test-renderer';
import { View } from 'react-native';

import InputContainer from '..';

describe('InputContainer', () => {
  it('renders with all props', () => {
    expect(
      renderer.create(
        <InputContainer
          containerStyle={{ backgroundColor: 'red' }}
          contentContainerStyle={{ backgroundColor: 'blue' }}
        >
          <View />
        </InputContainer>,
      ),
    ).toMatchSnapshot();
  });

  it('renders with minimum required props', () => {
    expect(renderer.create(<InputContainer />)).toMatchSnapshot();
  });
});
