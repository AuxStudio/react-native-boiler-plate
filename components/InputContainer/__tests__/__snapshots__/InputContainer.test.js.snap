// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`InputContainer renders with all props 1`] = `
<View
  style={
    Array [
      Object {
        "alignSelf": "stretch",
        "flex": 1,
      },
      Object {
        "backgroundColor": "red",
      },
    ]
  }
>
  <RCTScrollView
    automaticallyAdjustContentInsets={false}
    contentContainerStyle={
      Object {
        "backgroundColor": "blue",
      }
    }
    contentInset={
      Object {
        "bottom": 0,
      }
    }
    enableAutomaticScroll={true}
    enableResetScrollToCoords={true}
    extraHeight={75}
    extraScrollHeight={0}
    getScrollResponder={[Function]}
    handleOnScroll={[Function]}
    keyboardDismissMode="interactive"
    keyboardOpeningTime={250}
    keyboardShouldPersistTaps="handled"
    keyboardSpace={0}
    onScroll={[Function]}
    resetKeyboardSpace={[Function]}
    scrollEventThrottle={1}
    scrollForExtraHeightOnAndroid={[Function]}
    scrollIntoView={[Function]}
    scrollToEnd={[Function]}
    scrollToFocusedInput={[Function]}
    scrollToPosition={[Function]}
    showsVerticalScrollIndicator={true}
    update={[Function]}
    viewIsInsideTabBar={false}
  >
    <View>
      <View />
    </View>
  </RCTScrollView>
</View>
`;

exports[`InputContainer renders with minimum required props 1`] = `
<View
  style={
    Array [
      Object {
        "alignSelf": "stretch",
        "flex": 1,
      },
      undefined,
    ]
  }
>
  <RCTScrollView
    automaticallyAdjustContentInsets={false}
    contentInset={
      Object {
        "bottom": 0,
      }
    }
    enableAutomaticScroll={true}
    enableResetScrollToCoords={true}
    extraHeight={75}
    extraScrollHeight={0}
    getScrollResponder={[Function]}
    handleOnScroll={[Function]}
    keyboardDismissMode="interactive"
    keyboardOpeningTime={250}
    keyboardShouldPersistTaps="handled"
    keyboardSpace={0}
    onScroll={[Function]}
    resetKeyboardSpace={[Function]}
    scrollEventThrottle={1}
    scrollForExtraHeightOnAndroid={[Function]}
    scrollIntoView={[Function]}
    scrollToEnd={[Function]}
    scrollToFocusedInput={[Function]}
    scrollToPosition={[Function]}
    showsVerticalScrollIndicator={true}
    update={[Function]}
    viewIsInsideTabBar={false}
  >
    <View />
  </RCTScrollView>
</View>
`;
